fastdoubleparser-dev/src/main/java/ch.randelshofer.fastdoubleparser/ch/randelshofer/fastdoubleparser/AbstractNumberParser.java

```diff
@@ -116,10 +114,7 @@ protected static int lookupHex(byte ch) {
      * @param ch a character
      * @return the hex value or a value &lt; 0.
      */
-    protected static int lookupHex(char ch) {
-        // The branchy code is faster than the branch-less code.
-        // Branch-less code: return CHAR_TO_HEX_MAP[ch & 127] | (127 - ch) >> 31;
-        return ch < 128 ? CHAR_TO_HEX_MAP[ch] : -1;
+    static int lookupHex(char ch) {
+        return CHAR_TO_HEX_MAP[ch & 0xff];
```

Comment:
`         return CHAR_TO_HEX_MAP[ch & 0xff];`

This code incorrectly maps characters outside of the ASCII range. 
For example, the code incorrectly maps the character 'Ł' U+0141 to 'A' U+0041.

Reply:
You are right, I had to overlook something.